evalprompt/genRuleFromPrompt/testcase.mjs

```diff
@@ -0,0 +1,81 @@
+const { Spectral, Document } = require("@stoplight/spectral-core")
+import { alphabetical, casing, defined, enumeration, falsy, length, pattern, schema, truthy, unreferencedReusableObject, xor } from "@stoplight/spectral-functions";
+import { Yaml } from "@stoplight/spectral-parsers";
+import yaml from 'js-yaml';
+import * as fs from "node:fs";
+import * as path from "node:path";
+import { fileURLToPath } from "node:url";
+export default async function (output, context) {
+    const valueMap = {
+        alphabetical: alphabetical,
+        casing: casing,
+        defined: defined,
+        enumeration: enumeration,
+        falsy: falsy,
+        length: length,
+        pattern: pattern,
+        schema: schema,
+        truthy: truthy,
+        unreferencedReusableObject: unreferencedReusableObject,
+        xor: xor
+    };
+    const __dirname = path.dirname(fileURLToPath(import.meta.url));
+    const ruleYaml = output.match(/```yaml([\s\S]*?)```/)?.[1]?.trim();
+    const rule = yaml.load(ruleYaml);
+    for (let ymlObj in rule.rules) {
+        let obj = rule.rules[ymlObj];
+        if (obj.then) {
+            let thenObj = obj.then;
+            if (thenObj.function && thenObj.function in valueMap) {
+                thenObj.function = valueMap[thenObj.function];
+            }
+        }
+    }
+    const spectral = new Spectral();
+    spectral.setRuleset(rule);
+    hits = context.config.hits;
+    const componentsRes = [];
+    let TotalCount = 0;
+    for (let hit in hits) {
+        const file = hit;
+        TotalCount += hits[hit].length;
+        const myDocument = new Document(
+            fs.readFileSync(path.join(__dirname, "apispec", file), "utf-8").trim(),
+            Yaml,
+            "openapi.yaml",
+        );
+        let res = await spectral.run(myDocument);
+        let hitArr = hits[hit];
+        for (let item of res) {
+            let itemArr = item.path;
+            if (hitArr.some(r => JSON.stringify(r) === JSON.stringify(itemArr))) {
+                componentsRes.push({
+                    pass: true,
+                    score: 1,
+                    reason: `Hit the critical path ${itemArr} from file ${file}`,
+                });
+            } else {
+                componentsRes.push({
+                    pass: false,
+                    score: 0,
+                    reason: `Path ${itemArr} not expected from file ${file}`,
+                });
+            }
+        }
+    }
+    const passCount = componentsRes.filter(item => item.pass).length;
+    if (passCount != TotalCount || TotalCount != componentsRes.length) {
+        return {
+            pass: false,
+            score: passCount / componentsRes.length,
```

Comment:
If Total count is 10, and both passCount and componentsRes.length is 1, then the actual score is 1? But it is not expected.

Reply:
Update the score. now it becomes:
(Npass - Nnpass)/Chit , if Npass < Nnpass, then set to 0.